tuna/connection.py

```diff
@@ -241,36 +263,43 @@ def ssh_connect(self, abort=None):
     self.ssh = paramiko.SSHClient()
     self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
     for ssh_idx in range(NUM_SSH_RETRIES):
-      if abort is not None and chk_abort_file(self.id, self.logger):
-        self.logger.warning('Machine %s aborted ssh connection', self.id)
+      if abort is not None and chk_abort_file(id, self.logger):
+        self.logger.warning('Machine %s aborted ssh connection', id)
         return False
 
       try:
-        self.ssh.connect(self.hostname,
-                         username=self.user,
-                         password=self.password,
-                         port=self.port,
-                         timeout=SSH_TIMEOUT,
-                         allow_agent=False)
+        self.ssh.connect(  #type: ignore
+            self.hostname,  #type: ignore
+            username=self.user,  #type: ignore
+            password=self.password,  #type: ignore
+            port=self.port,  #type: ignore
+            timeout=SSH_TIMEOUT,
+            allow_agent=False)
```

Comment:
why specify type: ignore?

Reply:
connection.py: note: In member "ssh_connect" of class "Connection":
connection.py:283: error: "Connection" has no attribute "hostname" , to ignore this check from mypy.

Reply:
Are type annotations helpful if we escape them with ignores?
Also see line 55, these self.vars are declared, but may need an explicit declaration for this type checker to function properly.

Reply:
@technicalgrp89  these variables need to be initialized to None in the __init__ so pylint doesnt complain. We do this in every class in /tuna, you can check for reference